rust/src/alive_test/alive_test.rs

```diff
@@ -0,0 +1,324 @@
+    tx_ctl: Sender<AliveTestCtl>,
+) -> Result<(), AliveTestError> {
+    let mut count = 0;
+
+    if methods.contains(&AliveTestMethods::Icmp) {
+        for t in trgt.iter() {
+            count += 1;
+            let dst_ip = IpAddr::from_str(t).expect("Valid IP address");
+            let icmp = forge_icmp(dst_ip).expect("Valid ICMP packet");
+            let _ = alive_test_send_icmp_packet(icmp);
+        }
+    }
+    if methods.contains(&AliveTestMethods::TcpSyn) {
+        //unimplemented
+    }
+    if methods.contains(&AliveTestMethods::TcpAck) {
+        //unimplemented
+    }
+    if methods.contains(&AliveTestMethods::Arp) {
+        //unimplemented
+    }
+
+    tracing::debug!("Finished sending {count} packets");
+    sleep(Duration::from_millis(timeout)).await;
+    let _ = tx_ctl.send(AliveTestCtl::Stop).await;
+    Ok(())
+}
+
+impl Scanner {
+    pub fn new(target: Vec<Host>, methods: Vec<AliveTestMethods>, timeout: Option<u64>) -> Self {
+        Self {
+            target,
+            methods,
+            timeout,
+        }
+    }
+
+    pub async fn run_alive_test(&self) -> Result<(), AliveTestError> {
+        // TODO: Replace with a Storage type to store the alive host list
+        let mut alive = Vec::<(String, String)>::new();
+
+        if self.methods.contains(&AliveTestMethods::ConsiderAlive) {
+            for t in self.target.iter() {
+                alive.push((t.clone(), AliveTestMethods::ConsiderAlive.to_string()));
+                println!("{t} via {}", AliveTestMethods::ConsiderAlive.to_string())
+            }
+            return Ok(());
+        };
+
+        let capture_inactive = Capture::from_device("any")
+            .map_err(|e| AliveTestError::NoValidInterface(e.to_string()))?;
+        let trgt = self.target.clone();
+
+        let (tx_ctl, rx_ctl): (Sender<AliveTestCtl>, Receiver<AliveTestCtl>) = mpsc::channel(1024);
+        let (tx_msg, mut rx_msg): (Sender<AliveTestCtl>, Receiver<AliveTestCtl>) =
+            mpsc::channel(1024);
+
+        let capture_handle = tokio::spawn(capture_task(capture_inactive, rx_ctl, tx_msg));
+
+        let timeout = self.timeout.unwrap_or(DEFAULT_TIMEOUT_MS);
+        let methods = self.methods.clone();
+        let send_handle = tokio::spawn(send_task(methods, trgt, timeout, tx_ctl));
+
+        while let Some(AliveTestCtl::Alive {
+            ip: addr,
+            detection_method: method,
+        }) = rx_msg.recv().await
+        {
+            alive.push((addr.clone(), method.to_string()));
+            println!("{addr} via {method:?}");
+        }
+
+        match send_handle.await {
+            Ok(Ok(())) => (),
+            Ok(Err(e)) => return Err(e),
+            Err(e) => return Err(AliveTestError::JoinError(e.to_string())),
+        };
+        match capture_handle.await {
+            Ok(Ok(())) => (),
+            Ok(Err(e)) => return Err(e),
+            Err(e) => return Err(AliveTestError::JoinError(e.to_string())),
+        };
```

Comment:
I think it's safe enough to unwrap the `.await` call here, since the `JoinError` means that something failed within the task, so gracefully returning doesn't seem too useful.

I'd go with 

```suggestion rust
        send_handle.await.unwrap()?;
        capture_handle.await.unwrap()?;
```